language/stdlib/modules/approved_payment.move

```diff
@@ -0,0 +1,95 @@
+// Module that allows a payee to approve payments with a cryptographic signature. The basic flow is:
+// (1) Payer sends `metadata` to the payee
+// (2) Payee signs `metadata` and sends 64 byte signature back to the payer
+// (3) Payer sends an approved payment to the payee by sending a transaction invoking `deposit`
+//     with payment data + signature. The transaction will abort if the signature check fails.
+// Note: approved payment's are an accounting convenience/a courtesy mechansim for the payee, *not*
+// a hurdle that must be cleared for all payments to the payee. In addition, approved payments do
+// not have replay protection.
```

Comment:
I guess we decided to not include sender and/or amount and/or receiver, right?

Reply:
Correct, although we haven't ruled out signing `amount` in the future. Couple of thoughts on why not sender/receiver:

- Signing the sender address limits the flexibility of the payer to initiate payment from any of multiple addresses they may control
- Even though the receiver address is not signed, this approach ensures that an approved payment to address *A* must use the `ApprovedPayment::T` resource published under *A*'s address. That is: you can't do an approved payment to *A1* using an approval signature from *A2* (assuming they use different approval keys).